java/src/main/java/com/logicalclocks/hsfs/StorageConnector.java

```diff
@@ -58,8 +68,8 @@
     @JsonSubTypes.Type(value = StorageConnector.GcsConnector.class, name = "GCS"),
     @JsonSubTypes.Type(value = StorageConnector.BigqueryConnector.class, name = "BIGQUERY")
 })
-public abstract class StorageConnector {
 
+public abstract class StorageConnector extends StorageConnectorBase {
   @Getter @Setter
   protected StorageConnectorType storageConnectorType;
```

Comment:
Lots of duplication between `StorageConnector` and  `StorageConnectorBase`, especially variables. Also between child classes like `HopsFsConnector`.